src/mapping.ts

```diff
@@ -9,170 +9,158 @@ import {
-}
-
-/**
- * @notice sort tier as a string to fit schema enum reqs
- * @param _tier
- * @returns tier: string
- */
-function checkMergeClass(_tier: BigInt): string {
-  let tier: string;
-  tier = "ONE";
-  if (_tier == BigInt.fromI32(1)) {
-    return tier;
-  } else if (_tier == BigInt.fromI32(2)) {
-    tier = "TWO";
-  } else if (_tier == BigInt.fromI32(3)) {
-    tier = "THREE";
-  } else if (_tier == BigInt.fromI32(4)) {
-    tier = "FOUR";
-  }
-
-  return tier;
-}
-
-// /**
-//  * @notice object type where properties represent value-based-decoded nft metadata
-//  * TODO: not sure but I am getting a compile error here. Asked DK. I need to just look into this. This will be done in next PR as I clean up the code.
-//  */
-// type Values = {
-//   mass: BigInt;
-//   tier: string;
-//   color: string;
-// };
-
-// /**
-//  * @notice carries out decoding of value for a particular mass NFT
-//  * @returns value : Values
-//  * @dev TODO: see Object Type Values above and the error that is there. Once that is resolved then this can be used.
-//  */
-// function decodeValue(nftValue: BigInt): Values {
-//   let value: Values;
-//   let _tier: BigInt = nftValue / CLASS_MULTIPLIER;
-//   let tier = checkMergeClass(_tier);
-
-//   value = {
-//     mass: (nftValue % CLASS_MULTIPLIER) as BigInt,
-//     tier: tier,
-//     color: checkColor(tier),
-//   };
-
-//   return value;
-// }
+import { User, NFT, Collection } from "../generated/schema";
+import {
+  EMPTY_ADDRESS,
+  CLASS_MULTIPLIER,
+  NG_OMNIBUS,
+  MERGE_EXTADDR,
+  MERGE_NAME,
+  TOKENSTANDARD,
+  BIGINT_ZERO,
+  ADDR_DEAD,
+} from "./utils";
 
 /* ========== EVENT HANDLERS ========== */
 
 /**
  * @notice handles Transfer events and ultimately instantiates / updates User and NFT entities
  * @param event: Transfer
  * @dev all NFT entity generation is handled in other functions.
+ * Step 0 && Step 1: set NG_OMNIBUS as a whitelisted address as it is done in the constructor, and initialize new users
+ * Step 2a: either create new NFT if in minting phase
+ * Step 2b: OR updateNFT for all other Transfer scenarios incl: temporary transfers right before merging/burning, simple burning (direct to dead address, which then transfers to address(0))
+ * NOTE: MassUpdates will cover the merge scenario.
  */
 export function handleTransfer(event: Transfer): void {
   const to = event.params.to.toHex();
   const from = event.params.from.toHex();
 
   let user = User.load(to);
   const tokenId = event.params.tokenId.toString();
+  let collection = Collection.load(MERGE_EXTADDR);
+
+  if (!collection) {
+    collection = createCollection(MERGE_EXTADDR); // NEW for collection entity - first instance of minting
+  }
 
-  // Step 0 && Step 1: set NG_OMNIBUS as a whitelisted address as it is done in the constructor, and initialize new users
   if (!user) {
     user = new User(to);
-    if (NG_OMNIBUS) {
+    collection.totalUniqueOwnerAddresses++;
+    collection.save();
+    user.save();
```

Comment:
this can be deleted

Reply:
I'm guessing you are pointing to `user.save()` because it is repeated at the end of the entire function. If so, I have deleted this one!